packages/snaps-controllers/src/snaps/SnapController.test.ts

```diff
@@ -99,16 +102,34 @@ import {
   SNAP_APPROVAL_UPDATE,
 } from './SnapController';
 
-Object.defineProperty(globalThis, 'crypto', {
-  value: {
-    // eslint-disable-next-line @typescript-eslint/no-require-imports, @typescript-eslint/no-var-requires
-    ...require('node:crypto').webcrypto,
-    getRandomValues: jest.fn().mockReturnValue(new Uint32Array(32)),
-  },
-});
+if (!('CryptoKey' in globalThis)) {
+  // We can remove this once we drop Node 18
+  Object.defineProperty(globalThis, 'CryptoKey', {
+    value: webcrypto.CryptoKey,
+  });
+}
+
+globalThis.crypto ??= webcrypto as typeof globalThis.crypto;
+globalThis.crypto.getRandomValues = <Type extends ArrayBufferView | null>(
+  array: Type,
+) => {
+  if (array === null) {
+    return null as Type;
+  }
```

Comment:
Why is this function called with `null`? 🤔

Reply:
You tell me, I moved this from manageState where you wrote it like this 😅 https://github.com/MetaMask/snaps/pull/1741/commits/2c02f3c6f6977d8606ceaa5505d81172c4ffb718#diff-d864a03e4cd265bdacd771f8cb6fc5bb0fa327b2e0b918e50a77560bc5684e65R22-R26

Reply:
I realised that later in the review too. 😅 Does it work if you remove it, and use `array.fill(0)`? Not sure why I initially wrote it like this.

Reply:
The underlying type apparently expects the function to support `null` and `ArrayBufferView` which does not directly have `fill()`. So we may be forced to write it like this by the type def

Reply:
```
interface Crypto {
    /**
     * Available only in secure contexts.
     *
     * [MDN Reference](https://developer.mozilla.org/docs/Web/API/Crypto/subtle)
     */
    readonly subtle: SubtleCrypto;
    /** [MDN Reference](https://developer.mozilla.org/docs/Web/API/Crypto/getRandomValues) */
    getRandomValues<T extends ArrayBufferView | null>(array: T): T;
    /**
     * Available only in secure contexts.
     *
     * [MDN Reference](https://developer.mozilla.org/docs/Web/API/Crypto/randomUUID)
     */
    randomUUID(): `${string}-${string}-${string}-${string}-${string}`;
}
```